### PR TITLE
Fixing bug related to staying in the old list when clicking on another list

### PR DESCRIPTION
**Pull request by caiosba**
## Description

Fixing bug related to staying in the old list when clicking on another list.

Fixes: CV2-3743.

## Type of change

- [ ] Performance improvement and/or refactoring (non-breaking change that keeps existing functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Security mitigation or enhancement
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Automated test (add or update automated tests)

## How has this been tested?

I was not able to reproduce it locally, but I did see two GraphQL requests, one with the old ID and another one with the new ID. After this fix, I see only one GraphQL request, with the new ID.

## Things to pay attention to during code review

Please make sure that, besides checking that the bug was fixed, that other features still work: the current active list is highlighted, etc.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I've made sure my branch is runnable and given good testing steps in the PR description
- [x] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [ ] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I implemented any new components, they are self-contained, their `propTypes` are declared and they use React Hooks and, if data-fetching is required, they use Relay Modern with fragment containers
- [ ] If my components involve user interaction - specifically button, text fields, or other inputs - I have added a [BEM-like class name](https://meedan.atlassian.net/wiki/spaces/ENG/pages/1327628289/Naming+conventions+for+interactive+elements) to the element that is interacted with
- [ ] To the best of my knowledge, any new styles are applied according to the design system
- [ ] If I added a new external dependency, I included a rationale for doing so and an estimate of the change in bundle size (e.g., checked in https://bundlephobia.com/)

### CODE REVIEW

File: src/app/components/drawer/Projects/ProjectsComponent.js [hunk 1]
```diff
@@ -42,7 +42,6 @@ const ProjectsComponent = ({
   location,
 }) => {
   const [showNewListDialog, setShowNewListDialog] = React.useState(false);
-  const [collapsed, setCollapsed] = React.useState(false);
```
Comment (caiosba) [inline src/app/components/drawer/Projects/ProjectsComponent.js:45]: I realized this was not being used for anything.

File: src/app/components/drawer/Projects/ProjectsComponent.js [hunk 2]
```diff
@@ -61,7 +60,6 @@ const ProjectsComponent = ({
       setActiveItem({ type: path[2], id: parseInt(path[3], 10) });
     }
   }, [location.pathname]);
-  const isActive = (type, id) => type === activeItem.type && id === activeItem.id;
```
Comment (caiosba) [inline src/app/components/drawer/Projects/ProjectsComponent.js:64]: I think this was the line creating the problem... it creates derived-state that was triggering a second render.

**Review by dariusk — COMMENTED** [review]
I see a refactor of `isActive` (looks good) and a removal of `collapsed` and its event handler for the `onClick` ... does this break any functionality with collapse/expand? Or is that a feature we removed anyway?

File: src/app/components/drawer/Projects/ProjectsComponent.js [hunk 3]
```diff
@@ -71,15 +69,6 @@ const ProjectsComponent = ({
...
 
-  const handleClick = (route, id) => {
-    if (route !== activeItem.type || id !== activeItem.id) {
-      setActiveItem({ type: route, id });
-      if (collapsed) {
-        setCollapsed(false);
-      }
-    }
-  };
-
```
Comment (caiosba) [inline src/app/components/drawer/Projects/ProjectsComponent.js:82]: As I said above, the `collapsed` state was not being used for anything. And the `setActiveItem` was redundant with a `useEffect` in that component.

File: src/app/components/drawer/Projects/ProjectsComponent.js [hunk 4]
```diff
@@ -303,9 +292,8 @@ const ProjectsComponent = ({
                     routePrefix="list"
                     project={search}
                     teamSlug={team.slug}
-                    onClick={handleClick}
                     icon={search.is_part_of_feeds && <FeedIcon className={`${styles.listIcon} ${styles.listIconFeed}`} />}
-                    isActive={isActive('list', search.dbid)}
```
Comment (caiosba) [inline src/app/components/drawer/Projects/ProjectsComponent.js:308]: This was the actual fix.

**Comment by caiosba** [comment]
> I see a refactor of `isActive` (looks good) and a removal of `collapsed` and its event handler for the `onClick` ... does this break any functionality with collapse/expand? Or is that a feature we removed anyway?

Thanks @dariusk , even before seeing your comment I was adding PR comments to explain the changes - please let me know if they answer your questions!

**Review by dariusk — APPROVED** [review]
Looks good!

**Review by danielevalverde — APPROVED** [review]

**Pull request merged**